.github/workflows/release-porting-guide.yml

```
@@ -0,0 +1,85 @@
+---
+name: Ansible porting guide creation
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-build-data-branch:
+        description: >-
+          Release Branch name from the Ansible Build data PR.
+          (e.g. `refs/pull/1/merge`)
+        required: true
+      ansible-version:
+        description: >-
+          Exact release version. For example, 12.1.0
+        required: true
+
+jobs:
+  upload-porting-guide:
+    name: Extract the porting guide
+    runs-on: ubuntu-latest
+    env:
+      GIT_BRANCH: "release/porting-guide-${{ inputs.ansible-version }}"
+      ANSIBLE_VERSION_FULL: ${{ inputs.ansible-version }}
+      CI_COMMIT_MESSAGE: >-
+        Add the Ansible community ${{ inputs.ansible-version }} porting guide
+
+    steps:
+      - name: Extract the major version
+        run: echo "ANSIBLE_VERSION_MAJOR=${ANSIBLE_VERSION_FULL%%.*}" >> "${GITHUB_ENV}"
+        shell: bash --noprofile --norc -O extglob -eEuo pipefail {0}
+
+      - name: Check out this repo src
+        uses: actions/checkout@v4
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v4
+        with:
+          repository: ansible-community/ansible-build-data
+          ref: ${{ inputs.ansible-build-data-branch }}
+          path: ansible-build-data
+
+      - name: Copy the RST file to the correct path
+        run: >-
+          cp -v
+          "ansible-build-data/${{ env.ANSIBLE_VERSION_MAJOR }}/porting_guide_${{ env.ANSIBLE_VERSION_MAJOR }}.rst"
```

This is a shell command, so use env variables, not GHA interpolation:
```suggestion
          "ansible-build-data/${ANSIBLE_VERSION_MAJOR}/porting_guide_${ANSIBLE_VERSION_MAJOR}.rst"
```